internal/storage/append_batch.go

```diff
@@ -44,28 +48,31 @@ func (ab *AppendBatch) release() {
 func (ab *AppendBatch) SetLogEntry(llsn types.LLSN, glsn types.GLSN, data []byte) error {
 	dk := encodeDataKeyInternal(llsn, ab.dk)
 	ck := encodeCommitKeyInternal(glsn, ab.ck)
-	if err := ab.batch.Set(dk, data, nil); err != nil {
+	if err := ab.dataBatch.Set(dk, data, nil); err != nil {
 		return err
 	}
-	if err := ab.batch.Set(ck, dk, nil); err != nil {
+	if err := ab.commitBatch.Set(ck, dk, nil); err != nil {
 		return err
 	}
 	return nil
 }
 
 // SetCommitContext inserts a commit context.
 func (ab *AppendBatch) SetCommitContext(cc CommitContext) error {
-	return ab.batch.Set(commitContextKey, encodeCommitContext(cc, ab.cc), nil)
+	return ab.commitBatch.Set(commitContextKey, encodeCommitContext(cc, ab.cc), nil)
 }
 
 // Apply saves a batch of appended log entries to the storage.
 func (ab *AppendBatch) Apply() error {
-	return ab.batch.Commit(ab.writeOpts)
+	if err := ab.dataBatch.Commit(ab.writeOpts); err != nil {
+		return err
+	}
+	return ab.commitBatch.Commit(ab.writeOpts)
```

Comment:
In the case of an integrated DB, calling apply twice won't be degrade performance, right? I don't think there will be a big difference because it's only write in memtable.

Reply:
Yes, so I also think.